syntax = "proto2";
import "common/proto/config.proto";

package gdt.learning;

enum LogicalOperator {
  LOGICAL_OPERATOR_AND = 1;
  LOGICAL_OPERATOR_OR = 2;
};

enum MathOperator {
  MATH_OPERATOR_EQUAL = 1;
  MATH_OPERATOR_LESS_THAN = 2;
  MATH_OPERATOR_BIGGER_THAN = 3;
};

message FieldValueConfig {
  repeated string field_name = 1;
  optional FieldValue field_value = 2;
}

message Expression {
  optional MathOperator math_operator = 1;
  optional FieldValueConfig left_field = 2;
  optional FieldValueConfig right_field = 3;
};

message Lambda {
  optional bool not_operator = 1 [default = false];
  optional LogicalOperator logical_operator = 2;
  repeated Expression expression = 3;
  repeated Lambda lambda = 4;
};